### DISCUSSION TITLE
Ability to "Save Workspace"

### DISCUSSION THREAD

**Discussion by edhans** [TabularEditor/TabularEditor3#1086] [Ideas]
I find when working on a project, I get a pivot grid or two going, now the debugger, some DAX queries, etc.

It would be nice to be able to "Save the workspace" with all of that code and whatnot in one file (or folder) then when I start work the next day, I could open that workspace and all of my stuff comes back. I know I can do all of it one file at a time, but when you have 4-5 windows open, it is a little tedious.

**Comment by otykier** [comment]
Good suggestion. Just to clarify: Are you using Tabular Editor as an external tool for Power BI Desktop in this scenario? In that case, it might be a little difficult to find a good solution, as we have no way to know which data model a given set of "workspace files" belongs to.

Short term, we could probably allow saving pivot grids (just like DAX queries, DAX scripts and diagrams can be saved today), and allowing you to open multiple files (of different types) in one go, rather than one at a time.

But that doesn't solve the problem above. I.e. what should happen if you attempted to "open a workspace" containing a Pivot Grid, DAX query, etc. that was created for a different model than the currently open one?

**Comment by edhans** [comment]
My use case is 95% of the time it is in Power BI Desktop. 

I think the user should be responsible enough to not open the "Sales Analysis workspace" group of files in the "Manufacturing Forecast.PBIX" or tabular model, and if they did, then welcome to a bazillion syntax/intellisense errors.

In other words, I don't think it is unreasonable to require me to know where to open what workspace. I'd keep the names the same and in same folders (Sales Analysis workspace 1, Sales analysis workspace 2" for example) but don't think it should be on you to have some grand set of rules that keeps some sort of catalog for tabular models/PBIX files connected to workspace files.

**Comment by EthanEverett** [comment]
@otykier I would like to add support for the short term feature you proposed. My team and I work on specific models and being able to save a DAX query, Pivot Grid, diagram, etc. would allow us to more easily test new measures/relationships and verify the results are what we expected.

**Comment by edhans** [comment]
FWIW, My use case is now about 50% XMLA endpoint to premium, not Power BI desktop. Using TE3 is such a joy in these cases. 

Still think it should be on the user though to remember which "workspace files" to open with which dataset, and allow an option to close that workspace without resaving just in case it is the wrong one.

**Comment by Rofromthetrap** [comment]
Good day, Mr.@otykier ,

Is there any update on this ? This will definitely improve productivity and organisation, especially when working with multiple large projects.
